core/src/config.jl

```diff
@@ -65,9 +65,13 @@ for (T, kinds) in pairs(nodekinds)
 end
 const nodetypes = collect(keys(nodekinds))
 
+# A [] in a TOML is parsed as Vector{Union{}}
+Base.convert(::Type{Union{Float64, Vector{Float64}}}, x::Vector{Union{}}) =
+    convert(Vector{Float64}, x)
+
```

Comment:
I saw your comment in #gripes. I wonder if perhaps that's just a bug that needs to be filed?

In any case, this is type piracy. Can't you do https://github.com/Deltares/Ribasim/pull/504#discussion_r1296835149 again?

Reply:
I'm ok with this piracy, as the convert to Vector{Float64} is already defined in the same way, so this is just a custom union by us that we want to behave in the same way. Can't really see this having a negative impact.

Reply:
But `Configurations.from_dict` is meant for this right? It doesn't seem more complex.
I see Aqua tries to detect simple piracy these days, wonder why this isn't picked up.